components/cdc/src/endpoint.rs

```diff
@@ -573,15 +570,38 @@ impl<T: 'static + RaftStoreRouter<RocksEngine>> Endpoint<T> {
         }
     }
 
-    fn on_min_ts(&mut self, region_id: u64, min_ts: TimeStamp) {
-        if let Some(delegate) = self.capture_regions.get_mut(&region_id) {
-            if let Some(resolved_ts) = delegate.on_min_ts(min_ts) {
-                if resolved_ts < self.min_resolved_ts {
-                    self.min_resolved_ts = resolved_ts;
-                    self.min_ts_region_id = region_id;
+    fn on_min_ts(&mut self, regions: Vec<u64>, min_ts: TimeStamp) {
+        let mut resolved_regions = Vec::with_capacity(regions.len());
+        self.min_resolved_ts = TimeStamp::max();
+        for region_id in regions {
+            if let Some(delegate) = self.capture_regions.get_mut(&region_id) {
+                if let Some(resolved_ts) = delegate.on_min_ts(min_ts) {
+                    if resolved_ts < self.min_resolved_ts {
+                        self.min_resolved_ts = resolved_ts;
+                        self.min_ts_region_id = region_id;
+                    }
+                    resolved_regions.push(region_id);
                 }
             }
         }
+        self.broadcast_resolved_ts(resolved_regions);
+    }
+
+    fn broadcast_resolved_ts(&self, regions: Vec<u64>) {
+        let mut resolved_ts = ResolvedTs::default();
+        resolved_ts.regions = regions;
+        resolved_ts.ts = self.min_resolved_ts.into_inner();
+        let mut change_data_event = ChangeDataEvent::default();
+        change_data_event.set_resolved_ts(resolved_ts);
+        for (id, conn) in &self.connections {
+            if conn
+                .get_sink()
+                .send((0, change_data_event.clone()))
```

Comment:
0 is wrong, you can use [protobuf::Message::compute_size](https://tikv.github.io/doc/protobuf/trait.Message.html#tymethod.compute_size) to get its size.